### DISCUSSION TITLE
Monitoring Quarkus JVM Mode With Cryostat

### DISCUSSION THREAD

**Discussion by giscus** [quarkusio/quarkus#35567] [Quarkus Blog/Website]
# Monitoring Quarkus JVM Mode With Cryostat

Quarkus: Supersonic Subatomic Java

https://quarkus.io/blog/monitoring-quarkus-jvm-mode-with-cryostat/

**Comment by umutsahin** [comment]
Hi Andrew, can you also give more details about how to enable JMX on quarkus, even though I provide JMX related parameters to java it doesn't listen to the port I set. 

I found some old GitHub issues where team stated JMX is disabled on quarkus.
Thanks!

**Reply by andrewazores** [reply]
Hi @umutsahin, sorry for the late response, I only just happened to see your comment now.

I am not sure when the Quarkus team stated JMX is disabled. As far as I know it has always been possible to enable JMX on Quarkus in JVM mode, but it does need to be specifically enabled. Up until recently JMX and JFR were not available when building Quarkus in Native Image mode, so perhaps that is what they were referencing.

When you run your Quarkus container, do you see the JMX flags in the startup logs? Something like this:

```
$ podman run --rm quay.io/andrewazores/quarkus-test:latest
Starting the Java application using /opt/jboss/container/java/run/run-java.sh ...
INFO exec  java -Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager -Dcom.sun.management.jmxremote.port=9097 -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false -javaagent:/deployments/app/cryostat-agent.jar -XX:+UseParallelGC -XX:MinHeapFreeRatio=10 -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90 -XX:+ExitOnOutOfMemoryError -cp "." -jar /deployments/quarkus-run.jar
```
